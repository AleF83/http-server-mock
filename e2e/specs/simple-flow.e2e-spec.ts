import axios from 'axios';
import { CreateFakeServerRequest, RegisterResponseMockRequest } from 'http-server-mock-common';
import { createServerManagementClient, ServerManagementClient, FakeServerClient } from 'http-server-mock-client';

describe('Simple flow', () => {
  const name = 'e2e service mock';
  const port = 10000;

  let serverManagementClient: ServerManagementClient;
  let fakeServerClient: FakeServerClient;

  beforeAll(() => {
    serverManagementClient = createServerManagementClient('http://localhost:8080');
  });

  test('No servers on init', async () => {
    const servers = await serverManagementClient.listFakeServers();
    expect(servers).toHaveLength(0);
  });

  test('Create fake server', async () => {
    const fakeServerOptions: CreateFakeServerRequest = { name, port, startOnInit: false };
    fakeServerClient = await serverManagementClient.createFakeServer(fakeServerOptions);
    expect(fakeServerClient).toBeDefined();
  });

  test('Get fake-server info', async () => {
    const response = await fakeServerClient.getInfo();
    expect(response).toMatchSnapshot({
      id: expect.any(String),
    });
  });

  test('Start fake-server', async () => {
    await fakeServerClient.start();
  });

  test('Set response', async () => {
    const request: RegisterResponseMockRequest = {
      requestMatcher: {},
      expectedResponse: {},
    };
    const response = await fakeServerClient.registerResponseMock(request);
    expect(response).toBeDefined();
  });

  test('Call service', async () => {
    const response = await axios.get(`http://localhost:10000`);
    expect(response.status).toEqual(200);
    expect(response.data).toMatchSnapshot();
  });

  test('Stop fake-server', async () => {
    await fakeServerClient.stop();
  });

  test('Delete fake server', async () => {
    await serverManagementClient.deleteFakeServer(fakeServerClient.id);
  });
});
