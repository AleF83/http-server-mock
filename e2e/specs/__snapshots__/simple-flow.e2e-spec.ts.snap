// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple flow Call service 1`] = `
Object {
  "headers": Object {
    "accept": "application/json, text/plain, */*",
    "connection": "close",
    "host": "localhost:10000",
    "user-agent": "axios/0.19.2",
  },
  "method": "GET",
  "url": "/",
}
`;

exports[`Simple flow Get fake-server info 1`] = `
Object {
  "id": Any<String>,
  "name": "e2e service mock",
  "port": 10000,
  "status": "initialized",
}
`;
